### PR TITLE
Test files:scan for specific users (negative validation)

### PR DESCRIPTION
**Pull request by skshetry**
## Description

This adds a test for occ command `files:scan` for specific user.

## Related Issue

- Part of #33617

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by skshetry** [comment]
Negative validation for groups blocked by #33738

File: tests/acceptance/features/bootstrap/BasicStructure.php [hunk 1]
```diff
@@ -1676,21 +1676,61 @@ public function theAdministratorRequestsStatusPhp() {
...
+	 */
+	public function theAdministratorDeletesFileInLocalStorageUsingTheTestingApi($path) {
+		$user = $this->getAdminUsername();
+		$response = OcsApiHelper::sendRequest(
+			$this->getBaseUrl(),
+			$user,
+			$this->getAdminPassword(),
+			'DELETE',
+			"/apps/testing/api/v1/file",
+			['file' => LOCAL_STORAGE_DIR_ON_REMOTE_SERVER . "/$path"],
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/BasicStructure.php]: Currently, this function is unknown that there can be multiple local storages. At present, we only delete from the `local_storage`. Will change in upcoming PR.

File: tests/acceptance/features/bootstrap/BasicStructure.php [hunk 2]
```diff
@@ -1676,21 +1676,61 @@ public function theAdministratorRequestsStatusPhp() {
...
+	 * @When the administrator creates file :path with content :content in local storage :mountPoint using the testing API
+	 * @Given the administrator has created file :path with content :content in local storage :mountPount
+	 *
+	 * @param string $path
+	 * @param string $content
+	 * @param string $mountPoint
+	 *
+	 * @return void
+	 */
+	public function theAdministratorCreatesFileWithContentInLocalStorageUsingTheTestingApi($path, $content, $mountPoint) {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/BasicStructure.php]: please make the line shorter

File: tests/acceptance/features/bootstrap/BasicStructure.php [hunk 3]
```diff
@@ -1676,21 +1676,61 @@ public function theAdministratorRequestsStatusPhp() {
...
 		$user = $this->getAdminUsername();
 		$response = OcsApiHelper::sendRequest(
 			$this->getBaseUrl(),
 			$user,
 			$this->getAdminPassword(),
 			'POST',
 			"/apps/testing/api/v1/file",
-			['file' => LOCAL_STORAGE_DIR_ON_REMOTE_SERVER . "/$path", 'content' => $content],
+			['file' => TEMPORARY_STORAGE_DIR_ON_REMOTE_SERVER . "/$mountPoint/$path", 'content' => $content],
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/BasicStructure.php]: make line shorter

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 4]
```diff
@@ -592,11 +592,23 @@ public function theAdministratorHasChangedTheBackgroundJobsModeTo($mode) {
 	 *
 	 * @return void
 	 */
-	public function theAdministratorHasSetTheExternalStorageToBeNeverScannedUsingTheOccCommand() {
+	public function theAdministratorHasSetTheExternalStorageToBeNever() {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/OccContext.php]: I don't think we need that function at all, just use `theAdministratorHasSetTheExtStorageWithMountPoint()` and adjust the steps in feature files

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/cliMain/files.feature [hunk 5]
```diff
@@ -22,9 +22,9 @@ Feature: Files Operations command
   Scenario: Adding a file to local storage and running scan for a specific path should add files for only that path.
     Given using new DAV path
     And user "user0" has been created with default attributes
-    And the administrator has set the external storage to be never scanned automatically
-    And user "user0" has created folder "/local_storage/folder1"
-    And user "user0" has created folder "/local_storage/folder2"
+    And the administrator has set the external storage "local_storage" to be never scanned automatically
+    And user "user0" has created a folder "/local_storage/folder1"
```
Comment (phil-davis) [inline tests/acceptance/features/cliMain/files.feature]: ``created a folder`` diffs will need to be changed back to ``created folder``

**Review by individual-it — APPROVED** [review]

**Comment by skshetry** [comment]
Backport for stable10 on #33768

**Pull request merged**